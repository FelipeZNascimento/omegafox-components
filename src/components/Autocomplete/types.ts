export type TDropdownItem = {
  id: number;
  name: string;
  abbreviationEn: string;
};

export interface IAutocompleteProps {
  dropdownList: TDropdownItem[];
  onSelect: (item: TDropdownItem) => void;
}
