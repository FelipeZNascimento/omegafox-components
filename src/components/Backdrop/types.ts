export interface IBackdropProps {
  align?: 'left' | 'center' | 'right';
  children: JSX.Element;
  isOpen: boolean;
}
