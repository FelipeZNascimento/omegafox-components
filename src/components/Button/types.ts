export interface IButtonProps extends React.HTMLAttributes<HTMLButtonElement> {
  children?: React.ReactNode | null;
  icon?: React.ReactNode | null;
  isDisabled?: boolean;
  isDarkMode?: boolean;
  isSelected?: boolean;
  leftIcon?: boolean;
  size?: 'small' | 'medium' | 'large';
  shape?: 'rounded' | 'squared';
  shadowed?: boolean;
  tooltip?: string;
  variant?: 'primary' | 'danger' | 'neutral' | 'confirm' | 'warning';
  onClick: () => void;
}
