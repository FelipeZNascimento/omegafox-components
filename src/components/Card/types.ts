export interface ICardProps {
  isForceMobile?: boolean;
  isSelected: boolean;
  title: string;
  subtitle?: string | null;
  onClick?: null | (() => void);
  renderingStatusFunction?: null | (() => JSX.Element | null);
}
