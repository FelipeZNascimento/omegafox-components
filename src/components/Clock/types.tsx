import {
  FOOTBALL_MATCH_STATUS,
  AMERICAN_FOOTBALL_MATCH_STATUS
} from './constants';
import { TBetValues } from '../types';

type TClock = {
  time: number;
};

type FootballKeys = keyof typeof FOOTBALL_MATCH_STATUS;
export type TClockFootball = TClock & {
  status: typeof FOOTBALL_MATCH_STATUS[FootballKeys];
};

type AmericanFootballKeys = keyof typeof AMERICAN_FOOTBALL_MATCH_STATUS;
export type TClockAmericanFootball = TClock & {
  status: typeof AMERICAN_FOOTBALL_MATCH_STATUS[AmericanFootballKeys];
};

export interface IClockProps {
  betValue: TBetValues | null;
  clock: TClockAmericanFootball | TClockFootball;
  isMatchStarted?: boolean;
  isExpandable: boolean;
  isExpanded: boolean;
  isMatchEditable: boolean;
  timestamp: number;
}

export interface IClockMobileProps {
  clock: TClockAmericanFootball | TClockFootball;
  timestamp: number;
}
