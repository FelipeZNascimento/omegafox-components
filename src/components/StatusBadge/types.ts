export interface IStatusBadgeProps {
  color: 'green' | 'grey';
}
