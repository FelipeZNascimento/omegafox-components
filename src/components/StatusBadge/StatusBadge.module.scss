@use "../../styling/colors.scss" as colors;
@use "../../styling/color_utils.scss" as color_utils;

@import "../../styling/alignments.scss";
@import "../../styling/font-sizes.scss";
@import "../../styling/margins.scss";
@import "../../styling/paddings.scss";

.badge {
  border-radius: 50%;
  height: 8px;
  min-width: 8px;
  overflow: visible;
  position: relative;
  width: 8px;

  &Online {
    @extend .badge;
    background-color: colors.$mint;
    color: colors.$mint;

    &::after {
      position: absolute;
      top: 0;
      left: 0;
      width: 8px;
      height: 8px;
      border-radius: 50%;
      animation: ripple 2s infinite ease-in-out;
      border: 1px solid;
      content: "";
    }
  }

  &Offline {
    @extend .badge;

    background-color: colors.$grey3;
    color: colors.$red;
    opacity: 0.6;
  }
}

@keyframes ripple {
  0% {
    transform: scale(0.8);
    opacity: 1;
  }
  100% {
    transform: scale(2.4);
    opacity: 0;
  }
}
