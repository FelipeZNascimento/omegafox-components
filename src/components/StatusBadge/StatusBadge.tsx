import React from 'react';

import classNames from 'classnames';
import { IStatusBadgeProps } from './types';
import styles from './StatusBadge.module.scss';

export const StatusBadge = ({ color }: IStatusBadgeProps) => {
  const onlineBadgeClass = classNames({
    [styles.badgeOnline]: color === 'green',
    [styles.badgeOffline]: color === 'grey'
  });

  return <div className={onlineBadgeClass} />;
};
