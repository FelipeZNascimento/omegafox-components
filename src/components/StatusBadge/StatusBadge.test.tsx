import React from 'react';
import { render } from '@testing-library/react';
import { StatusBadge } from './StatusBadge';
import { IStatusBadgeProps } from './types';

describe('<StatusBadge />', () => {
  const renderComponent = ({ color }: IStatusBadgeProps) =>
    render(<StatusBadge color={color} />);

  it('should render', () => {
    const { container } = renderComponent({ color: 'green' });

    expect(container).toMatchSnapshot();
  });
});
