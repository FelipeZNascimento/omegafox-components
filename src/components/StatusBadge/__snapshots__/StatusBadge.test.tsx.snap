// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StatusBadge /> should render 1`] = `
<div>
  <div
    class="badgeOnline"
  />
</div>
`;
